### PR TITLE
Add auto-column grid classes

### PR DESCRIPTION
**Pull request by nolder3193**
Add classes to generate auto-column grid templates based on min-width px value

### CODE REVIEW

File: theme.js [hunk 1]
```diff
@@ -6,7 +8,20 @@ const spacing = [
   90, 95, 96, 100
 ];
 
-// https://uicolors.app/edit?sv1=primary:50-ecebff/100-dedbff/200-c7c2ff/300-a899ff/400-937aff/500-7e61ff/600-6a4ce1/700-5b33c7/800-3e238b/900-251452/950-11082b;slate:50-f4f4fb/100-e7e7f4/200-d9d9ed/300-c4c4e3/400-aeaed6/500-9692c4/600-8178b0/700-655b95/800-4b4276/900-332b5a/950-1d1641;red:50-fff1f3/100-ffe3e6/200-ffccd4/300-ffa1b1/400-ff708a/500-f93a63/600-e7174e/700-c30d41/800-a30e3d/900-8b103b/950-4e031b;yellow:50-fff9eb/100-fef0c8/200-fddd8b/300-fccb4f/400-fbbd23/500-f5af00/600-f1a200/700-e58e00/800-c87200/900-9c4f00/950-7a3700;green:50-edfcf5/100-d4f7e4/200-abedcd/300-78ddb3/400-3dd198/500-1bc58a/600-0eb47d/700-0d9c71/800-0f805e/900-0c644b/950-085441;blue:50-f0f8fe/100-deeefb/200-c4e2f8/300-9bd0f3/400-6db6ed/500-549ee7/600-4687dd/700-3778d2/800-2c5ca5/900-213e6e/950-192e4d
+// Generates auto-fit and auto-fill grids based on a minimum column width suffix (in pixels)
+let autoColumns = [];
+
```
Comment (jclusso) [inline theme.js:13]: I would eliminate the line between these to keep them togather.

File: theme.js [hunk 2]
```diff
@@ -6,7 +8,20 @@ const spacing = [
...
+for (let i = 1; i < 51; i++) {
```
Comment (jclusso) [inline theme.js:14]: Reads cleaner to use <=.
```suggestion
for (let i = 1; i <= 50; i++) {
```

File: theme.js [hunk 3]
```diff
@@ -125,6 +140,7 @@ export const theme = {
         'Monaco', 'DejaVu Sans Mono', 'Lucida Console'
       ]
     },
+    gridTemplateColumns: { ...Object.fromEntries(columns) },
```
Comment (jclusso) [inline theme.js:143]: I believe you meant:
```suggestion
    gridTemplateColumns: { ...Object.fromEntries(autoColumns) },
```

File: theme.js [hunk 4]
```diff
@@ -6,7 +8,20 @@ const spacing = [
...
+let autoColumns = [];
+
+for (let i = 1; i < 51; i++) {
+  autoColumns.push([
+    `auto-fit-${i * 10}`,
+    `repeat(auto-fit, minmax(min(100%, ${(i * 10) / 16}rem), 1fr))`
+  ]);
+
+  autoColumns.push([
+    `auto-fill-${i * 10}`,
```
Comment (jclusso) [inline theme.js:21]: I'm wondering if they should be `fit-10` and `fill-10`. Was looking at [this post](https://github.com/tailwindlabs/tailwindcss/pull/8812#issuecomment-1200487353) while reviewing this.

**Review by jclusso — CHANGES_REQUESTED** [review]
I'm hesitant to merge this for a few reasons. 

1. Do we really need 50 of each of these? It seems rather excessive to need `auto-fit/fill-10` through `auto-fit/fill-500`. How many of these are currently used?
2. Is this needed enough to be included in all Emailable projects?

File: theme.js (hunk 4 above)
Comment (nolder3193) [reply theme.js:21]: Oh yeah good find. Makes sense.

**Comment by nolder3193** [comment]
Suppose it doesn't _have_ to be in both but I feel like I often have to use very specific values to make a layout's items fit nicely and/or not wrap, etc. I would normally use arbitrary values for stuff like this but the fact that you have to write out so damn much makes it inconvenient (`grid-cols-[repeat(auto-fit,minmax(min(100%,15rem),1fr))]` for example). I have definitely used values like `190` in the past which is why I went with that granularity.

**Comment by jclusso** [comment]
@nolder3193 my thought is lets add this in the site and add it here once we actually need it.

**Comment by jclusso** [comment]
Closing for now and we can reopen in future.

**Pull request closed without merging**